### DISCUSSION TITLE
[question] Why autocompleter has a min-chars-limit?

### DISCUSSION THREAD

**Discussion by capric98** [searxng/searxng#1082] [Ideas]
Hello, after using searxng for a while I just noticed a slight difference between searxng and Google about the auto-complete, that searxng has a 4 min-chars-limit which means it will not work until `len($query) >= 4`.
https://github.com/searxng/searxng/blob/27f8fa6fe066c15dfc60bc2c50533ef87663ce29/searx/static/themes/simple/src/js/main/search.js#L71

I am not familiar with js so I did a simple search and it seems like seaxng uses `autocompletejs/autocomplete.js`? And it has `MinChars: 0` by [default](https://github.com/autocompletejs/autocomplete.js/blob/87069524f3b95e68f1b54d8976868e0eac1b2c83/src/autocomplete.ts#L125).

So is it possible to disable this limit from js side or move it to `settings.yml`? The limit may help to avoid abuse, but for personal usage, disabling it may result in a better experience IMO.

**Comment by mrpaulblack** [comment]
For example the google autocomplete backend does not do rate limiting or IP blocking at all. So changing the minChars to 0 would not mean that an instance would be more prone to getting blocked by the autocomplete services. It would just add more requests to the instance. For that reason we could even try to just set it by default to 0 instead of the current 4 min chars. I think the instances can handle the additional traffic tbh. IF you want to can you do an PR and change the value to 0? We can discuss this further there :+1: 

[EDIT] Also to answer your question. I am not sure, but I also think its probably to save bandwidth and to work against abuse...

**Comment by capric98** [comment]
https://github.com/searxng/searxng/pull/1087

**Reply by return42** [reply]
For reference https://github.com/searxng/searxng/pull/1091
